monomer_test.go

```diff
@@ -1,19 +1,213 @@
+	cometHeader := header.ToComet()
+
+	require.Equal(t, &bfttypes.Header{
+		ChainID: header.ChainID.String(),
+		Height:  int64(header.Height),
+		Time:    time.Unix(int64(header.Time), 0),
+		AppHash: header.StateRoot.Bytes(),
+	}, cometHeader)
+}
+
+func TestToEth(t *testing.T) {
+	header := newTestHeader()
+	ethHeader := header.ToEth()
+
+	require.Equal(t, &ethtypes.Header{
+		ParentHash:      header.ParentHash,
+		Root:            header.StateRoot,
+		Number:          big.NewInt(int64(header.Height)),
+		GasLimit:        header.GasLimit,
+		MixDigest:       common.Hash{},
+		Time:            header.Time,
+		UncleHash:       ethtypes.EmptyUncleHash,
+		ReceiptHash:     ethtypes.EmptyReceiptsHash,
+		BaseFee:         common.Big0,
+		WithdrawalsHash: &ethtypes.EmptyWithdrawalsHash,
+		Difficulty:      common.Big0,
+	}, ethHeader)
+}
+
+func TestBlockNewBlock(t *testing.T) {
+	block := monomer.NewBlock(newTestHeader(), bfttypes.Txs{})
+	ethBlock, err := block.ToEth()
+	require.NoError(t, err)
+
+	newBlock, err := monomer.SetHeader(block)
+	require.NoError(t, err)
+	require.Equal(t, ethBlock.Hash(), newBlock.Header.Hash)
+
+	_, err = monomer.SetHeader(nil)
+	require.Error(t, err)
+}
+
+func TestBlockSetHeader(t *testing.T) {
+	block := monomer.NewBlock(newTestHeader(), bfttypes.Txs{})
+	ethBlock, err := block.ToEth()
+	require.NoError(t, err)
+
+	newBlock, err := monomer.SetHeader(block)
+	require.NoError(t, err)
+	require.Equal(t, ethBlock.Hash(), newBlock.Header.Hash)
+
+	_, err = monomer.SetHeader(nil)
+	require.Error(t, err)
+}
+
+func TestBlockMakeBlock(t *testing.T) {
+	header := newTestHeader()
+	emptyTxs := bfttypes.Txs{}
+
+	block := monomer.NewBlock(header, emptyTxs)
+	ethBlock, err := block.ToEth()
+	require.NoError(t, err)
+
+	newBlock, err := monomer.MakeBlock(header, emptyTxs)
+	require.NoError(t, err)
+
+	require.Equal(t, ethBlock.Hash(), newBlock.Header.Hash)
+}
+
 func TestBlockToEth(t *testing.T) {
 	l1InfoTx, depositTx, cosmosEthTx := testutils.GenerateEthTxs(t)
-	block := testutils.GenerateBlockFromEthTxs(t, l1InfoTx, []*ethtypes.Transaction{depositTx}, []*ethtypes.Transaction{cosmosEthTx})
+	block := testutils.GenerateBlockFromEthTxs(t,
+		l1InfoTx,
+		[]*ethtypes.Transaction{depositTx},
+		[]*ethtypes.Transaction{cosmosEthTx},
+	)
+
+	ethTxs, err := rolluptypes.AdaptCosmosTxsToEthTxs(block.Txs)
+	require.NoError(t, err)
+
 	ethBlock, err := block.ToEth()
 	require.NoError(t, err)
-	for i, ethTx := range []*ethtypes.Transaction{l1InfoTx, depositTx, cosmosEthTx} {
-		require.EqualExportedValues(t, ethTx, ethBlock.Body().Transactions[i])
+	require.EqualExportedValues(t, ethtypes.NewBlockWithWithdrawals(
+		block.Header.ToEth(),
+		ethTxs,
+		nil,
+		[]*ethtypes.Receipt{},
+		[]*ethtypes.Withdrawal{},
+		trie.NewStackTrie(nil),
+	), ethBlock)
+
+	newBlock := monomer.NewBlock(newTestHeader(), bfttypes.Txs{[]byte("transaction1"), []byte("transaction2")})
+	_, err = newBlock.ToEth()
+	require.Error(t, err)
```

Comment:
can we add a comment describing what this case is testing? It may be even better to separate this out into a separate test like `TestBlockToEthFailsWithoutDepositTxs` (long names are ok in tests; the test name should generally describe what the test does so we don't need comments)